modules/generic_worker/templates/run-generic-worker.sh.erb

```diff
@@ -22,38 +22,67 @@ exitstatus="${PIPESTATUS[0]}"
+#        terminate.
+# 71     The worker was terminated via an interrupt signal (e.g. Ctrl-C pressed).
+# 72     The worker is running on spot infrastructure in AWS EC2 and has been served a
+#        spot termination notice, and therefore has shut down.
+# 73     The config provided to the worker is invalid.
+# 74     Could not grant provided SID full control of interactive windows stations and desktop.
+# 75     Not able to create an ed25519 key pair.
+# 76     Not able to save generic-worker config file after fetching it from AWS provisioner
+#        or Google Cloud metadata.
+# 77     Not able to apply required file access permissions to the generic-worker config
+#        file so that task users can't read from or write to it.
 
-        # create a semaphore file
-        echo "quarantined worker" > <%= scope.lookupvar('users::builder::home') %>/quarantined
+# If worker is in bad state, reboot it first time. If after reboot the worker continue to be
+# in bad state, quarantine the worker and stop reboot it
+    if [ -f <%= scope.lookupvar('users::builder::home') %>/reboot_semaphore ]; then
+        # The worker was rebooted once
+        if [ -f <%= scope.lookupvar('users::builder::home') %>/quarantined ]; then
+            echo "Worker is quarantined"
+        else
+            echo "Quarantine worker"
+            # quarantine the worker using github.com/mozilla-platform-ops/quarantine-worker CLI
+            /usr/local/bin/quarantine-worker releng-hardware "<%= @worker_type %>" "<%= @worker_group %>" "<%= @hostname %>" 720h5s
 
-        # Submit a new bug to Bugzilla that reports that the worker has been quarantined, 
-        # in order to alert CI Duty to the failure, track status and facilitate resolution.
-        create_bug_self_failure
+            # create a semaphore file
+            echo "quarantined worker" > <%= scope.lookupvar('users::builder::home') %>/quarantined
+        fi
+    else 
+        # first time reboot, create the reboot semaphore file and reboot the machine
+        date > <%= scope.lookupvar('users::builder::home') %>/reboot_semaphore
+        <%= @reboot_command %>
+        /bin/sleep 120
     fi
-elif [ $exitstatus -gt 69 -o $exitstatus -lt 69 ]; then
+else 
     # If exitstatus is different by 69 (grater than 69 or lower than 69)
     # The worker was removed from the quarantin and run with success a job
     # Remove quarantined file, if file exist
     if [ -f <%= scope.lookupvar('users::builder::home') %>/quarantined ]; then
         /bin/rm -rf <%= scope.lookupvar('users::builder::home') %>/quarantined
     fi
+    # Remove reboot semaphore file
+    if [ -f <%= scope.lookupvar('users::builder::home') %>/reboot_semaphore ]; then
+        /bin/rm -rf <%= scope.lookupvar('users::builder::home') %>/reboot_semaphore
+    fi
     # Self resolve the bug and remove bug_id file if file exist
     if [ -f <%= scope.lookupvar('users::builder::home') %>/bug_id ]; then
```

Comment:
if you aren't creating bugs then there is no need to resolve them or remove their semaphore